packages/fuchsia_remote_debug_protocol/lib/src/common/logging.dart

```diff
@@ -0,0 +1,149 @@
+///
+/// Verbosity is increasing from zero (none) to four (fine). The fifth level
+/// (all) logs everything.
+enum LoggingLevel {
+  /// Logs no logs.
+  none,
+
+  /// Logs severe messages at the most (note that severe messages are always
+  /// logged).
+  ///
+  /// Severe means that the process has
+  /// encountered a critical level of failure in which it cannot recover and
+  /// will terminate as a result.
+  severe,
+
+  /// Logs warning messages at the most.
+  ///
+  /// Warning implies that an error was encountered, but the process will
+  /// attempt to continue, and may/may not succeed.
+  warning,
+
+  /// Logs info messages at the most.
+  ///
+  /// An info message is for determining information about the state of the
+  /// application as it runs through execution.
+  info,
+
+  /// Logs fine logs at the most.
+  ///
+  /// A fine message is one that is not important for logging outside of
+  /// debugging potential issues in the application.
+  fine,
+
+  /// Logs everything.
+  all,
+}
+
+/// Signature of a function that logs a [LogMessage].
+typedef void LoggingFunction(LogMessage log);
+
+/// The default logging function.
+///
+/// Runs the [print] function using the format string:
+///   '[${log.levelName}]::${log.tag}--${log.time}: ${log.message}'
+///
+/// Exits with status code 1 if the `log` is [LoggingLevel.severe].
+LoggingFunction defaultLoggingFunction = (LogMessage log) {
+  print('[${log.levelName}]::${log.tag}--${log.time}: ${log.message}');
+  if (log.level == LoggingLevel.severe) {
+    exit(1);
+  }
+};
+
+/// Represents a logging message created by the logger.
+///
+/// Includes a message, the time the message was created, the level of the log
+/// as an enum, the name of the level as a string, and a tag. This class is used
+/// to print from the global logging function defined in
+/// [Logger.loggingFunction] (a function that can be user-defined).
+class LogMessage {
+  /// The actual log message.
+  final String message;
+
+  /// The time the log message was created.
+  final DateTime time;
+
+  /// The level of this log.
+  final LoggingLevel level;
+
+  /// The human readable level of this log.
+  final String levelName;
+
+  /// The tag associated with the message. This is set to [Logger.tag] when
+  /// emitted by a [Logger] object.
+  final String tag;
+
+  /// Creates a log, including the level of the log, the time it was created,
+  /// and the actual log message.
+  ///
+  /// When this message is created, it sets its [time] to [DateTime.now].
+  LogMessage(this.message, this.tag, this.level)
+      : this.levelName =
+            level.toString().substring(level.toString().indexOf('.') + 1),
+        this.time = new DateTime.now();
+}
+
+/// Logs messages using the global [LoggingFunction] and logging level.
+///
+/// Example of setting log level to [LoggingLevel.warning] and creating a
+/// logging function:
+///
+/// Logger.globalLevel = LoggingLevel.warning;
```

Comment:
wrap examples in triple backtick blocks, tagged as "dart", as in:
```
'''dart
// example here
'''
```
...but using backticks instead of apostrophes

Reply:
Done.